examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,225 @@
+    const fieldName = hasFieldNameChanged ? field!.name : ""
+    const placeholder = isUnsupported ? "Unsupported Field" : fieldConfig.name
+
+    const isDisabled = useMemo(() => isUnsupported || isIgnored, [isUnsupported, isIgnored])
+
+    return (
+        <Fragment>
+            <div
+                className="source-field"
+                aria-disabled={isDisabled}
+                onClick={() => onFieldToggle(fieldConfig.field!.id)}
+                role="button"
+            >
+                <input
+                    type="checkbox"
+                    disabled={isUnsupported}
+                    checked={!isDisabled}
+                    onChange={e => {
+                        e.stopPropagation()
+                    }}
+                />
+                <span>{fieldConfig.name}</span>
+            </div>
+            <svg xmlns="http://www.w3.org/2000/svg" width="8" height="16">
+                <path
+                    d="M 3 11 L 6 8 L 3 5"
+                    fill="transparent"
+                    strokeWidth="1.5"
+                    stroke="#999"
+                    strokeLinecap="round"
+                    strokeLinejoin="round"
+                ></path>
+            </svg>
+            <input
+                type="text"
+                style={{
+                    width: "100%",
+                    opacity: isDisabled ? 0.5 : 1,
+                }}
+                disabled={isDisabled}
+                placeholder={placeholder}
+                value={fieldName || ""}
+                onChange={e => {
+                    if (!field) return
+                    if (!e.target.value.trim()) {
+                        onFieldNameChange(field.id, fieldConfig.name)
+                    } else {
+                        onFieldNameChange(field.id, e.target.value)
+                    }
+                }}
+            />
+        </Fragment>
+    )
+}
+
+export function FieldMapping({
+    savedFieldsConfig,
+    existingFields,
+    dataSource,
+    savedSlugFieldId,
+    onSubmit,
+}: {
+    savedFieldsConfig: FieldConfig[] | undefined
+    existingFields: ManagedCollectionField[]
+    dataSource: DataSource
+    savedSlugFieldId: string | null
+    onSubmit: (dataSource: DataSource, fields: FieldConfig[], slugFieldId: string) => Promise<void>
+}) {
+    const [fields, setFields] = useState<FieldConfig[]>(
+        savedFieldsConfig ?? computeFieldConfigs(existingFields, dataSource)
+    )
+    const [disabledFieldIds, setDisabledFieldIds] = useState<Set<string>>(
+        new Set(savedFieldsConfig?.filter(field => field.isNew).map(field => field.field!.id))
+    )
+
+    const possibleSlugFields = useMemo(
+        () =>
+            fields
+                .map(field => field.field)
+                .filter(isNotNull)
+                .filter(field => !disabledFieldIds.has(field.id) && field.type === "string"),
+        [fields, disabledFieldIds]
```

Comment:
I think this would be easier to read if the inline function would use curly braces